src/server/transport.rs

```diff
@@ -68,21 +68,37 @@ pub trait RaftStoreRouter: Send + Clone {
 #[derive(Clone)]
 pub struct ServerRaftStoreRouter {
     pub ch: SendCh<StoreMsg>,
+    store_id: u64,
 }
 
 impl ServerRaftStoreRouter {
-    pub fn new(ch: SendCh<StoreMsg>) -> ServerRaftStoreRouter {
-        ServerRaftStoreRouter { ch: ch }
+    pub fn new(ch: SendCh<StoreMsg>, store_id: u64) -> ServerRaftStoreRouter {
+        ServerRaftStoreRouter {
+            ch: ch,
+            store_id: store_id,
+        }
+    }
+
+    fn validate_msg(&self, msg: &StoreMsg) -> RaftStoreResult<()> {
+        if let StoreMsg::RaftMessage(ref m) = *msg {
```

Comment:
`RaftCmd` should be checked too. I prefer moving this check to `send_raft_msg` and `send_command` respectively, so that here doesn't need to match again.